rules/Naming/ExpectedNameResolver/InflectorSingularResolver.php

```diff
@@ -106,4 +100,83 @@ private function singularizeCamelParts(string $currentName): string
 
         return $resolvedName;
     }
+
+    // see https://gist.github.com/peter-mcconnell/9757549
+    private function singularize(string $word): string
+    {
+        $singular = [
+            '/(quiz)zes$/i' => '\\1',
+            '/(matr)ices$/i' => '\\1ix',
+            '/(vert|ind)ices$/i' => '\\1ex',
+            '/^(ox)en/i' => '\\1',
+            '/(alias|status)es$/i' => '\\1',
+            '/([octop|vir])i$/i' => '\\1us',
+            '/(cris|ax|test)es$/i' => '\\1is',
+            '/(shoe)s$/i' => '\\1',
+            '/(o)es$/i' => '\\1',
+            '/(bus)es$/i' => '\\1',
+            '/([m|l])ice$/i' => '\\1ouse',
+            '/(x|ch|ss|sh)es$/i' => '\\1',
+            '/(m)ovies$/i' => '\\1ovie',
+            '/(s)eries$/i' => '\\1eries',
+            '/([^aeiouy]|qu)ies$/i' => '\\1y',
+            '/([lr])ves$/i' => '\\1f',
+            '/(tive)s$/i' => '\\1',
+            '/(hive)s$/i' => '\\1',
+            '/([^f])ves$/i' => '\\1fe',
+            '/(^analy)ses$/i' => '\\1sis',
+            '/((a)naly|(b)a|(d)iagno|(p)arenthe|(p)rogno|(s)ynop|(t)he)ses$/i' => '\\1\\2sis',
+            '/([ti])a$/i' => '\\1um',
+            '/(n)ews$/i' => '\\1ews',
+            '/s$/i' => '',
+        ];
```

Comment:
How does this compete with inflector setup? Some difference comparison would be helpful :+1:

Reply:
tbh, I am not convinved whether rector should do this pluralization/singularization stuff at all.

but at least its now reduced to a minimum